### PR TITLE
Implemented core functionality for Subreddit

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: lib/src/exceptions.dart [hunk 1]
```diff
@@ -6,29 +6,43 @@
...
 /// Thrown due to a error on the side of the client due to incorrect integration of DRAW.
 class DRAWClientError implements Exception {
   DRAWClientError(this.message);
-  String message;
+  final String message;
   String toString() =>
       'DRAWClientError: $message This is likely due to issues with your ini file.';
 }
+
+class DRAWRedirectResponse implements Exception {
```
Comment (ckartik) [inline lib/src/exceptions.dart]: Should we call this Exception type: DRAWRedirectResponseError instead?

File: lib/src/draw_config_context.dart [hunk 2]
```diff
@@ -167,6 +167,7 @@ class DRAWConfigContext {
     String authorizeUrl,
     String siteName = 'default',
   }) {
+    return;
```
Comment (ckartik) [inline lib/src/draw_config_context.dart]: Was there something not working with the Config object?

File: lib/src/listing/listing_generator.dart [hunk 3]
```diff
@@ -34,30 +34,34 @@ abstract class ListingGenerator {
     final kLimitKey = 'limit';
     final kAfterKey = 'after';
     final nullLimit = 1024;
-    final paramsInternal = params ?? new Map();
+    final paramsInternal = (params != null) ? new Map.from(params) : new Map();
```
Comment (ckartik) [inline lib/src/listing/listing_generator.dart]: logic is a bit hard to read, should we put:
`final paramsInternal = (params == null) ? new Map() : new Map.from(params);`

File: lib/src/exceptions.dart (hunk 1 above)
Comment (bkonyi) [reply lib/src/exceptions.dart]: This isn't an error, but is still an exception, so we should leave it as is.

File: lib/src/draw_config_context.dart (hunk 2 above)
Comment (bkonyi) [reply lib/src/draw_config_context.dart]: Ah, yes, something was broken, can't remember what.

File: lib/src/listing/listing_generator.dart (hunk 3 above)
Comment (bkonyi) [reply lib/src/listing/listing_generator.dart]: Done.

**Pull request merged**